tests/__init__.py

```diff
@@ -83,18 +95,36 @@ def build(self, options: Sequence[str] = (), args: Sequence[str] = ()) -> Comple
         )
 
     def boot(self, options: Sequence[str] = (), args: Sequence[str] = ()) -> CompletedProcess:
-        return self.mkosi("boot", [*options, "--debug"], args, stdin=sys.stdin if sys.stdin.isatty() else None)
+        result = self.mkosi(
+            "boot",
+            [*options, "--debug"],
+            args, stdin=sys.stdin if sys.stdin.isatty() else None,
+            check=False,
+        )
+
+        if result.returncode != 123:
+            raise subprocess.CalledProcessError(result.returncode, result.args, result.stdout, result.stderr)
+
+        return result
 
     def qemu(self, options: Sequence[str] = (), args: Sequence[str] = ()) -> CompletedProcess:
-        return self.mkosi(
+        result = self.mkosi(
             "qemu",
             [*options, "--debug"],
             args,
             stdin=sys.stdin if sys.stdin.isatty() else None,
             user=INVOKING_USER.uid,
             group=INVOKING_USER.gid,
+            check=False,
         )
 
+        rc = 0 if self.distribution == Distribution.ubuntu or self.distribution.is_centos_variant() else 123
```

Comment:
Thinking some more about this, is this the right way 'round? Qemu runs on the host and not in the image, no?

Reply:
Yes? This is about systemd versions, not qemu